chain/client/src/stateless_validation/partial_witness_tracker.rs

```diff
@@ -133,8 +140,8 @@ impl PartialEncodedStateWitnessTracker {
         let chunk_hash = partial_witness.chunk_header().chunk_hash();
         let entry = self.parts_cache.get_mut(&chunk_hash).unwrap();
 
-        if let Some(_witness) = entry.insert_in_cache_entry(partial_witness, rs) {
-            // TODO(stateless_validation): Send encoded state witness to client here.
+        if let Some(encoded_witness) = entry.insert_in_cache_entry(partial_witness, rs) {
+            self.client_sender.send(ProcessChunkStateWitnessMessage(encoded_witness));
```

Comment:
One additional check pending. It's now totally possible that if we are the chunk producer as well as the validator, we would receive parts for our own chunk state witness.

I don't yet have the check to ignore that. Will add in either this PR or a future one.

Reply:
Note that we do already handle bypassing validation of our own state witness and sending chunk endorsement directly to block producer.